sql/src/main/java/io/crate/expression/symbol/FuncArg.java

```diff
@@ -36,11 +36,16 @@
     DataType valueType();
 
     /**
-     * Indicates whether a Symbol can be casted or not.
-     * Typically, we only allow casting of Literals.
+     * Indicates whether a Symbol could be casted or not (it expresses
```

Comment:
"From the symbol's perspective it can be casted. " - ah you're right. It makes more sense when looking at it from the symbol's perspective. 

I'm in two minds about the `allowEagerCast`. Looking at it from the Symbol's perspective I think `canBeCasted` and the "convertibility" test are sufficiently expressive for now.